backend/coins/btc/account.go

```diff
@@ -234,6 +236,31 @@ func (account *Account) gapLimits(
 	return limits, nil
 }
 
+// getMinRelayFeeRate fetches the min relay fee from the server and returns it. The value is cached
+// so that subsequent calls are instant. This is important as this function can be called many times
+// in succession when validating tx proposals.
+func (account *Account) getMinRelayFeeRate() btcutil.Amount {
+	unlock := account.RLock()
+	cached := account.minRelayFeeRate
+	unlock()
+	if cached != nil {
+		return *cached
+	}
+
+	doneCh := make(chan struct{})
+	account.coin.Blockchain().RelayFee(func(feeRatePerKb btcutil.Amount) {
+		defer account.Lock()()
+		account.minRelayFeeRate = &feeRatePerKb
+		close(doneCh)
+	}, func(error) {})
+	<-doneCh
+
+	unlock = account.RLock()
+	defer unlock()
+	account.log.Infof("min relay fee rate: %s", *account.minRelayFeeRate)
+	return *account.minRelayFeeRate
```

Comment:
Too many lock/unlocks imho. What is returned here might not match `account.coin.Blockchain().RelayFee(...)` result, in theory. In practice, like you said it's stable so doesn't matter much. It just leaves this feeling of not being entirely consistent.

If the next bitcoin core server version somehow makes the fee more dynamic, this might become an issue, in addition to the cache above.

I would always simply unlock at the method return. So, only one lock/unlock. The callers of this menthod always block and wait for the result anyway. But it would require a separate sync.Mutex though. Wouldn't want to lock the whole account.

Reply:
Totally right! Simplified it with a separate mutex. PTAL.

Reply:
Much nicer! :+1: :pray: